sale_discount_code/views/sale_order.xml

```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
+    <data>
+<record id="view_order_form" model="ir.ui.view">
+    <field name="model">sale.order</field>
+    <field name="inherit_id" ref="sale.view_order_form"/>
+    <field name="arch" type="xml">
+        <field name="pricelist_id" position="after">
+            <field name="discount_code"/>
```

Comment:
the field should be just before the total

![image](https://user-images.githubusercontent.com/1164578/30863681-ef465648-a2d1-11e7-8e0f-d389f018dca1.png)